kpops/components/base_components/kafka_connect.py

```diff
@@ -241,7 +241,7 @@ def template(self, api_version: str, ca_file: str, cert_file: str) -> None:
             self.name,
             KafkaConnectorType.SOURCE,
             {},
-            # offset_topic=self.offset_topic,
+            # offset_topic=self.offset_topic, # mypy did not like this
```

Comment:
Why is that?

Reply:
The comment? mypy throws the following when the `offset_topic` is not commented out:
```
kpops/components/base_components/kafka_connect.py:240: error: Unexpected
keyword argument "offset_topic" for "_get_kafka_connect_resetter_values" of
"KafkaConnector"  [call-arg]
            values = self._get_kafka_connect_resetter_values(
                     ^
Found 1 error in 1 file (checked 101 source files)
```

Reply:
yeah, the usage of `kwargs` seems a little weird to me. Cann we change it to `**kwargs` (dict unpacking) everywhere in this file? then the error should go away

Reply:
wdyt @raminqaf ?

```diff
diff --git a/kpops/components/base_components/kafka_connect.py b/kpops/components/base_components/kafka_connect.py
index a47d482..16c07e3 100644
--- a/kpops/components/base_components/kafka_connect.py
+++ b/kpops/components/base_components/kafka_connect.py
@@ -151,7 +151,7 @@ class KafkaConnector(PipelineComponent, ABC):
         )
 
         stdout = self.__install_connect_resetter(
-            trimmed_name, connector_name, connector_type, dry_run, kwargs
+            trimmed_name, connector_name, connector_type, dry_run, **kwargs
         )
 
         if dry_run and self.helm_diff.config.enable:
@@ -176,7 +176,7 @@ class KafkaConnector(PipelineComponent, ABC):
         connector_name: str,
         connector_type: KafkaConnectorType,
         dry_run: bool,
-        kwargs,
+        **kwargs,
     ) -> str:
         return self.helm.upgrade_install(
             release_name=release_name,
@@ -192,7 +192,7 @@ class KafkaConnector(PipelineComponent, ABC):
             values=self._get_kafka_connect_resetter_values(
                 connector_name,
                 connector_type,
-                kwargs,
+                **kwargs,
             ),
         )
 
@@ -200,7 +200,7 @@ class KafkaConnector(PipelineComponent, ABC):
         self,
         connector_name: str,
         connector_type: KafkaConnectorType,
-        kwargs,
+        **kwargs,
     ) -> dict:
         return {
             **KafkaConnectResetterValues(
@@ -240,8 +240,7 @@ class KafkaSourceConnector(KafkaConnector):
         values = self._get_kafka_connect_resetter_values(
             self.name,
             KafkaConnectorType.SOURCE,
-            {},
-            # offset_topic=self.offset_topic, # mypy did not like this
+            offset_topic=self.offset_topic,
         )
         stdout = self.helm.template(
             self._get_release_name(self.name),
@@ -287,7 +286,7 @@ class KafkaSinkConnector(KafkaConnector):
     def template(self, api_version: str, ca_file: str, cert_file: str) -> None:
         flags = HelmTemplateFlags(api_version, ca_file, cert_file, self.version)
         values = self._get_kafka_connect_resetter_values(
-            self.name, KafkaConnectorType.SINK, {}
+            self.name, KafkaConnectorType.SINK
         )
         stdout = self.helm.template(
             self._get_release_name(self.name),
```

Reply:
@disrupted this works, should I commit it like you suggested?

Reply:
@disrupted does it make sense to type the `**kwargs`?

Reply:
@sujuka99 yep
@raminqaf which type annotation would you suggest?

Reply:
I pushed it like that, will revert it if needed